### ISSUE TITLE
proposal: `unnecessary_super`

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/linter#5020]
## Description

Don't use `super.` to invoke a method which is not overridden by the current class.

## Details

An unnecessary `super.<member>` invocation can lead to bugs if an override of the
member is later  added to the current class.

**good:**

**bad:**

## Kind

Guards against errors, promotes good style.

cc @munificent should we add this to the style guide?

## Bad Examples

```dart
class A {
  int get x => 1;
}

class  B extends A {
  int get y => super.x;
}
```

## Good Examples 

```dart
class A {
  int get x => 1;
}

class  B extends A {
  int get y => x;
}
```

## Discussion

@nshahan came upon a situation recently when trying to merge a subtype with its supertype, there were now some invalid calls to `super.<member>`. These never needed to be super invocations - there was no override. Most likely, they were leftovers from a refactor at some point, or possibly a copy/paste from a different subtype.

### Discussion checklist

- [x] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [x] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [x] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [x] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Writing Lints]: https://github.com/dart-lang/linter/blob/main/doc/writing-lints.md
[Effective Dart]: https://dart.dev/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[SDK Tracker]: https://github.com/dart-lang/sdk/issues
